src/fees/hooks.ts

```diff
@@ -45,6 +51,58 @@ export function usePaidFees(fees: Fee[]) {
 }
 
 export function useMaxSendAmount(
+  tokenId: string,
+  comment?: string
+): { loading: true; maxAmount: null } | { loading: false; maxAmount: BigNumber } {
+  const token = useTokenInfo(tokenId)
+  const balance = token?.balance ?? new BigNumber(0)
+  const feeCurrencies = useFeeCurrencies(token?.networkId)
+  const walletAddress = useSelector(walletAddressSelector)
+  const { result: prepareTransactionsResult, loading } = useAsync(async () => {
+    try {
+      if (!walletAddress) {
+        // should never happen
+        throw new Error('Cannot calculate max send amount because wallet address is undefined')
+      }
+      if (!token) {
+        // should never happen
+        throw new Error(
+          `Cannot calculate max send amount because token with tokenId ${tokenId} not found`
+        )
+      }
+      const output = await prepareSendTransactions({
+        amount: balance,
+        token,
+        recipientAddress: walletAddress, // dummy recipient
+        walletAddress,
+        feeCurrencies,
+        comment,
+      })
+      if (!output) {
+        throw new Error(
+          'prepareSendTransactions returned undefined. Can occur when balance is 0 or if token has no address and is not native'
+        )
+      }
+      return output
+    } catch (error) {
+      Logger.error(TAG, 'Error calculating max send amount', error)
+      return
+    }
+  }, [token])
+  if (loading) {
+    return { loading: true, maxAmount: null }
+  }
+  if (prepareTransactionsResult?.type === 'need-decrease-spend-amount-for-gas') {
```

Comment:
I noticed that this doesn't seem to ever happen for sends-- it might only work right now for transactions where we include a gas estimate rather than querying for it from Viem